### ISSUE TITLE
Graylog 1.3: Search fails if begins with square bracket and includes wildcard

### ISSUE THREAD

**Issue by mwhidden** [Graylog2/graylog2-server#1971]
### Problem description
Example log entry: `2016-03-23 22:06:17,500 WARN [org.mybatis.spring.mapper.ClassPathMapperScanner] (ServerService Thread Pool -- 64) No MyBatis mapper was found in '[com.mycorp.im.mappers]' package. Please check your configuration.`

I'd like to search for log entries where the class name begins with `[org.mybatis`.

Searching for `\[org.mybatis` returns no results ("Nothing found in stream...")
Searching for `\[org.mybatis*` returns no results ("Nothing found in stream...")
`\[org.mybatis.spring.mapper.ClassPathMapperScanner\]` works fine.
So does `\[org.mybatis.spring.mapper.ClassPathMapperScanner`,
But not `\[org.mybatis.spring.mapper.ClassPathMapperScanne`, ("Nothing found in stream...")
nor `\[org.mybatis.spring.mapper.ClassPathMapperScanne*`, ("Nothing found in stream...")

### Steps to reproduce the problem
See above

### Environment

* Graylog Version: 1.3
* Elasticsearch Version: 1.7.4
* MongoDB Version:
* Operating System: RedHat
* Browser version: Firefox 44.0.2

**Comment by kroepke** [comment]
Unfortunately the way elasticsearch works these kinds of queries are tricky to understand:

Each input is analyzed, meaning the string is split into "terms". You can see which terms the field was broken down to in the message detail view.
Search queries are also analyzed, however certain queries prevent analysis, one of them is wildcard search (because it's not clear to elasticsearch which "words" are complete).

Basically you can only search for "words" that are broken into terms right now.

**Comment by mwhidden** [comment]
When I try to view the terms "Show Terms of Message", I get an InternalServerError "Could not load field terms. Loading field terms failed with status: Internal Server Error" but I don't find anything in the server or -web log files, even with DEBUG level logging enabled.

However, the admin was able to view the terms, and it's not so tricky as you made it sound -- it looks like punctuation with whitespace on either side is elided from search terms. 

So looks like a "feature" not a bug. I've been there...
